src/Functions/readWkt.cpp

```diff
@@ -125,19 +126,17 @@ class FunctionReadWKTCommon : public IFunction
 
         auto discriminators_column = ColumnVariant::ColumnDiscriminators::create();
 
-        auto try_deserialize_type = [&](const std::function<void()> & func, WKTTypes type) -> bool
+        auto try_deserialize_type = [&] (const std::function<void()> & deserialize_func, const String & data, const String & target_prefix, WKTTypes type) -> bool
         {
-            try
+            auto lower_data = boost::to_lower_copy(data);
```

Comment:
There are three problems:
1. We do lowercase many times instead of just one.
2. We do it for all the string, which can be large, instead of just for the type name.
3. We do string comparison multiple times, while we can do it in a single pass.